onnx_chainer/replace_func.py

```diff
@@ -0,0 +1,129 @@
+import inspect
+from inspect import Parameter
+
+import chainer
+
+
+class WrappedFunctionNode(chainer.FunctionNode):
+
+    def __init__(self, name, func, args, kwargs, attributes=None):
+        self.user_name = name
+        self.func = func
+        self.args = args
+        self.kwargs = kwargs
+
+        if attributes is not None:
+            self.set_attributes(attributes)
+
+    def set_attributes(self, attributes):
+        for p in attributes:
+            if isinstance(p, tuple):
+                assert isinstance(p[0], int)
+                setattr(self, p[1], self.args[p[0]])
+            elif isinstance(p, str):
+                setattr(self, p, self.kwargs.get(p, None))
+
+    def forward(self, xs):
+        self.xs = xs
+        results = self.func(*self.args, **self.kwargs)
+        if isinstance(results, (tuple, list)):
+            dummy_results = tuple(
+                ret.array if _isvar(ret) else ret for ret in results)
```

Comment:
[optional]  It seems `_isvar` is often used in ternary ops. How about defining
```
def _unwrap_var(a):
  return a.array if _isvar(a, chainer.Variable) else a
```
?

Reply:
fixed